compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -928,16 +948,51 @@ impl<'interner> Monomorphizer<'interner> {
             }
             DefinitionKind::Global(global_id) => {
                 let global = self.interner.get_global(*global_id);
-
-                let expr = if let GlobalValue::Resolved(value) = global.value.clone() {
-                    value
-                        .into_hir_expression(self.interner, global.location)
-                        .map_err(MonomorphizationError::InterpreterError)?
+                let id = global.id;
+                let name = definition.name.clone();
+                if let Some(seen_global) = self.globals.get(&id) {
```

Comment:
The case for `DefinitionKind::Global` is quite long now, can you break it up into its own function?

Reply:
done